src/tests/plans/D3MAaveTypeBufferPlan.t.sol

```diff
@@ -0,0 +1,158 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import { D3MPlanBaseTest, DaiLike } from "./D3MPlanBase.t.sol";
+import { D3MAaveTypeBufferPlan } from "../../plans/D3MAaveTypeBufferPlan.sol";
+
+contract ADaiMock {
+
+    address public dai;
+
+    constructor(address _dai) {
+        dai = _dai;
+    }
+
+    function UNDERLYING_ASSET_ADDRESS() external view returns (address) {
+        return dai;
+    }
+
+}
+
+contract DaiMock {
+
+    uint256 liquidity;
+
+    function setLiquidity(uint256 value) external {
+        liquidity = value;
+    }
+
+    function balanceOf(address) external view returns (uint256) {
+        return liquidity;
+    }
+
+}
+
+contract D3MAaveTypeBufferPlanTest is D3MPlanBaseTest {
+
+    ADaiMock adai;
+    DaiMock  _dai;
+
+    D3MAaveTypeBufferPlan plan;
+
+    event Disable();
+
+    function setUp() public override {
+        contractName = "D3MAaveTypeBufferPlan";
+
+        dai = DaiLike(address(_dai = new DaiMock()));
+        adai = new ADaiMock(address(dai));
+
+        vm.expectEmit(true, true, true, true);
+        emit Rely(address(this));
+        d3mTestPlan = address(plan = new D3MAaveTypeBufferPlan(address(adai)));
+    }
+
+    function test_constructor() public {
+        assertEq(address(plan.dai()), address(dai));
+        assertEq(address(plan.adai()), address(adai));
+    }
+
+    function test_file() public {
+        checkFileUint(d3mTestPlan, contractName, ["buffer"]);
+    }
+
+    function test_auth_modifier() public {
+        plan.file("buffer", 1);
+        plan.deny(address(this));
+
+        checkModifier(d3mTestPlan, "D3MAaveTypeBufferPlan/not-authorized", [
+            abi.encodeWithSelector(D3MAaveTypeBufferPlan.disable.selector)
+        ]);
+    }
+
+    function test_implements_getTargetAssets() public override {
+        _dai.setLiquidity(40 ether);
+        plan.file("buffer", 100 ether);
+        assertEq(plan.getTargetAssets(0), 60 ether);
+    }
+
+    function test_increase_liquidity() public {
+        plan.file("buffer", 100 ether);
+        assertEq(plan.getTargetAssets(20 ether), 120 ether);
+    }
+
+    function test_decrease_liquidity_sole_provider() public {
+        plan.file("buffer", 100 ether);
+        assertEq(plan.getTargetAssets(0), 100 ether);
+        _dai.setLiquidity(100 ether);   // Simulate adding liquidity
```

Comment:
Mmm not very critical but this returning to liquidity of 100 each time just to change it the next line seems a bit redundant.

Reply:
Yeah I know, but it separates the high-level actions such as "D3M adds liquidity", "User borrows", etc. Any issue if I leave it?